hopsworks-common/src/main/java/io/hops/hopsworks/common/python/library/LibraryInstaller.java

```diff
@@ -0,0 +1,420 @@
+      allCondaCommands.sort(ASC_COMPARATOR);
+      for (final CondaCommands cc : allCondaCommands) {
+        try {
+          try {
+            switch (cc.getOp()) {
+              case CREATE:
+                createNewImage(cc);
+                break;
+              case INSTALL:
+                installLibrary(cc);
+                break;
+              case UNINSTALL:
+                uninstallLibrary(cc);
+                break;
+              case REMOVE:
+                deleteImage(cc);
+                break;
+              case EXPORT:
+                exportLibraries(cc);
+                break;
+              default:
+                throw new UnsupportedOperationException("conda command unknown: " + cc.getOp());
+            }
+          } catch (Throwable ex) {
+            LOG.log(Level.WARNING, "Could not execute command with ID: " + cc.getId(), ex);
+            commandsController.updateCondaCommandStatus(
+                cc.getId(), CondaStatus.FAILED, cc.getInstallType(), cc.getMachineType(),
+                cc.getArg(), cc.getProjectId(), cc.getUserId(), cc.getOp(), cc.getLib(), cc.getChannelUrl(),
+                cc.getArg());
+            continue;
+          }
+          commandsController.updateCondaCommandStatus(
+              cc.getId(), CondaStatus.SUCCESS, cc.getInstallType(), cc.getMachineType(),
+              cc.getArg(), cc.getProjectId(), cc.getUserId(), cc.getOp(), cc.getLib(), cc.getChannelUrl(),
+              cc.getArg());
+        } catch (ServiceException ex) {
+          LOG.log(Level.WARNING, "Could not update command with ID: " + cc.getId());
+        }
+      }
+    } finally {
+      schedule();
+    }
+  }
+
+  private void createNewImage(CondaCommands cc) throws IOException {
+    File baseDir = new File("/tmp/docker/" + cc.getProjectId().getName());
+    baseDir.mkdir();
+    File dockerFile = new File(baseDir, "dockerFile_" + cc.getProjectId().getName());
+    BufferedWriter writer = new BufferedWriter(new FileWriter(dockerFile));
+    writer.write("FROM " + projectUtils.getFullDockerImageName(cc.getDockerImage()));
+    writer.close();
+
+    String prog = settings.getSudoersDir() + "/dockerImage.sh";
+
+    ProcessDescriptor processDescriptor = new ProcessDescriptor.Builder()
+        .addCommand("/usr/bin/sudo")
+        .addCommand(prog)
+        .addCommand("create")
+        .addCommand(dockerFile.getAbsolutePath())
+        .addCommand(settings.getRegistry() + "/" + cc.getProjectId().getName())
+        .redirectErrorStream(true)
+        .setCurrentWorkingDirectory(baseDir)
+        .setWaitTimeout(60L, TimeUnit.SECONDS)
+        .build();
+
+    try {
+      ProcessResult processResult = osProcessExecutor.execute(processDescriptor);
+      if (processResult.getExitCode() != 0) {
+        String errorMsg = "Could not create the docker image. Exit code: " + processResult.getExitCode()
+            + " Error: " + processResult.getStdout();
+        LOG.log(Level.SEVERE, errorMsg);
```

Comment:
We don't need to log it here, it will be logged by https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-api/src/main/java/io/hops/hopsworks/api/exception/mapper/RESTApiThrowableMapper.java

Reply:
All exceptions that are in a REST endpoint stack trace, go through the ThrowableMapper

Reply:
Except that this is asynchronous so not returned in a REST endpoint.

Reply:
This is asynchronous so not returned in a REST endpoint.
I will rework on this part to replace the part previously handled by Kibana.